Content.Client/GameObjects/EntitySystems/ItemTeleporterSystem.cs

```diff
@@ -0,0 +1,71 @@
+using Content.Client.GameObjects.Components.Movement;
+using Content.Shared.GameObjects.Components.Movement;
+using Content.Client.Interfaces.GameObjects;
+using Content.Shared.Input;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects.EntitySystems;
+using Robust.Client.Interfaces.Graphics.ClientEye;
+using Robust.Client.Interfaces.Input;
+using Robust.Client.Player;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Input;
+using Robust.Shared.IoC;
+
+namespace Content.Client.GameObjects.EntitySystems
+{
+    [UsedImplicitly]
+    public class ItemTeleporterSystem : EntitySystem
```

Comment:
Why can't all of this be handled on the server? This is quite a lot of fragile code.

Reply:
Yeah you're right. I ended up copying the ranged weapon system but that was dumb in hindsight as they're different use cases.

I'm trying to use IAttack but I can't seem to get it working.

Reply:
You want `IAfterAttack` or `IAttack` instead. `IAttack` is used for when the mob is in combat mode (which isn't 100% implemented yet).

I only now realize the ridiculous naming here jeez.